src/OrchardCore.Modules/OrchardCore.Media/Assets/js/MediaApp/fileupload.js

```diff
@@ -12,6 +12,21 @@ $(document).on('mediaApp:ready', function () {
                 { name: '__RequestVerificationToken', value: antiForgeryToken },
             ]
         },
+        change: function (e, data) {
+            const maxSizeLimit = parseInt($('#maxSizeLimit').val());
+            var exceededFilesList = [];
+
+            $.each(data.files, function (index, file) {
+                if (file.size > maxSizeLimit) {
+                    exceededFilesList.push(index);
+                    mediaApp.errors.push(`The file '${file.name}' exceeds the limit.`);
+                }
+            });
+
+            for (var i = 0; i < exceededFilesList.length; i++) {
+                data.files.splice(i, 1);
```

Comment:
I think this is not going to work. "i" should be "exceededFilesList[i]".
But even with that modification it is not going to work, because when you remove an item from an array the index you had recorded in the previous loop is not valid anymore.
Maybe you could try creating a new array with valid files, and then do : 
`data.files = validFiles;`